cogs/file_cog.py

```diff
@@ -0,0 +1,141 @@
+
+                heif_file = pyheif.read(file)
+                image = Image.frombytes(
+                    heif_file.mode,
+                    heif_file.size,
+                    heif_file.data,
+                    "raw",
+                    heif_file.mode,
+                    heif_file.stride,
+                )
+                with io.BytesIO() as image_binary:
+                    image_binary = await save_jpg_with_target_size(
+                        image, image_binary, 8e6
+                    )
+                    if image_binary is None:
+                        await message.channel.send("`ERROR: failed to resize image`")
+
+                    await message.channel.send(
+                        file=discord.File(
+                            fp=image_binary, filename=attachment.filename[:-5] + ".png"
+                        )
+                    )
+        return
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, payload):
+        yimin_emote = self.bot.get_emoji(733516365243220038)
+        if payload.user_id == self.bot.user.id or payload.emoji != yimin_emote:
+            return
+
+        channel = self.bot.get_channel(payload.channel_id)
+        message = await channel.fetch_message(payload.message_id)
+
+        if message.attachments == 0:
+            return
+
+        dpi = 200  # dots per inch
+        for attachment in message.attachments:
+            if any(attachment.filename.endswith(i) for i in [".pdf", ".PDF"]):
+                print(f"Converting {attachment.filename} to images...")
+
+                async with aiohttp.ClientSession() as session:
+                    async with session.get(attachment.url) as r:
+                        pdf_file = await r.content.read()
+
+                info = pdfinfo_from_bytes(pdf_file, userpw=None, poppler_path=None)
+
+                max_pages = info["Pages"]
+                pages_in_mem = 1
+                for pg in range(1, max_pages + 1, pages_in_mem):
+                    pages = convert_from_bytes(
+                        pdf_file,
+                        dpi,
+                        first_page=pg,
+                        last_page=min(pg + pages_in_mem - 1, max_pages),
+                    )
+                    for idx, page in enumerate(pages):
+                        with io.BytesIO() as image_binary:
+                            page.save(image_binary, "PNG")
+                            image_binary.seek(0)
+                            await message.channel.send(
+                                file=discord.File(
+                                    fp=image_binary,
+                                    filename=f"{os.path.splitext(attachment.filename)}_{idx + 1}.png",
+                                )
+                            )
+
+
+def setup(bot):
+    # Adds cog to bot from main.py
+    if (
+        config.STAGE != "dev"
+    ):  # dont add this cog if we are in dev to avoid duplication, remove for testing
```

Comment:
looks like blackd did some weirdness with the formatting because you put the comment on the same line. Try putting the comment on the next line so it's more readable. Great solution for only adding it to staging or prod though. 👌 
```suggestion
    if config.STAGE != "dev":
        # dont add this cog if we are in dev to avoid duplication, remove for testing
```